pkg/webhook/patch.go

```diff
@@ -158,6 +159,48 @@ func addVolumeMount(pod *corev1.Pod, mount corev1.VolumeMount) patchOperation {
 	return patchOperation{Op: "add", Path: path, Value: value}
 }
 
+func addEnvVars(pod *corev1.Pod, app *v1beta2.SparkApplication) []patchOperation {
+	var envVars []corev1.EnvVar
+	var containerName string
+	if util.IsDriverPod(pod) {
+		envVars = app.Spec.Driver.Env
+		containerName = config.SparkDriverContainerName
+	} else if util.IsExecutorPod(pod) {
+		envVars = app.Spec.Executor.Env
+		containerName = config.SparkExecutorContainerName
+	}
+
+	i := 0
+	// Find the driver or executor container in the pod.
+	for ; i < len(pod.Spec.Containers); i++ {
+		if pod.Spec.Containers[i].Name == containerName {
+			break
+		}
+	}
+	basePath := fmt.Sprintf("/spec/containers/%d/env", i)
+
+	var value interface{}
+	var patchOps []patchOperation
+
+	first := false
+	if len(pod.Spec.Containers[i].Env) == 0 {
+		first = true
+	}
+
+	for _, envVar := range envVars {
```

Comment:
This reminds me that we are **not** doing the correct thing for `addTolerations` and maybe some others. We should fix those.

Reply:
Created https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/pull/697 to fix those.